### PR TITLE
#638 Add Store doc

### PR DESCRIPTION
**Pull request by freddyli7**
Add store and database doc

### CODE REVIEW

File: docs/core/store.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<!--
+order: 3
+-->
+
+# Store and Database
+
+Learn about the store and database concept on Ethermint. {synopsis}
+
+Since Ethermint uses the Cosmos-SDK as an ABCI application, when starting an Ethermint node,
+a Key-Value database will be initialized and all Ethermint blockchain data will be persisted into the node machine.
```
Comment (noot) [inline docs/core/store.md]: maybe say node host machine?

File: docs/core/store.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+Store is the abstract data structure in Ethermint to organize and persist blockchain data into a physical disk.
+
+There are two kinds of store in Ethermint:  
+
+1. Cosmos SDK MultiStore.
+2. Ethermint CommitStateDB.
+
+Cosmos SDK MultiStore is used in most Ethermint modules such as bank, governance and staking. The synchronization of this store on the network is taken care of by [Tendermint](https://github.com/tendermint/tendermint).
+
+Different store prefix(store key) will be assigned to each Ethermint module since they all implement the cosmos-sdk MultiStore interface. Cosmos MultiStore can be accessed by `sdk.Context` expect `ethermint/evm` module.
```
Comment (noot) [inline docs/core/store.md]: `A unique store prefix (also referred to as the store key) wil...`

File: docs/core/store.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+For `evm` module, Ethermint creates a type called `CommitStateDB` which implements `ethvm.StateDB` from go-ethereum lib, and it aims to be compatible with ethereum EVM transactions.
```
Comment (noot) [inline docs/core/store.md]: ` implements the `ethvm.StateDB` interface from go-ethereum. It aims to be...`

**Review by noot — COMMENTED** [review]
looks good, a couple small recommendations

File: docs/core/store.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<!--
+order: 3
+-->
+
+# Store and Database
+
+Learn about the store and database concept on Ethermint. {synopsis}
```
Comment (araskachoi) [inline docs/core/store.md:7]: should we add links to learn more about the cosmos sdk modules?
https://docs.cosmos.network/master/core/store.html

File: docs/core/store.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+1. Cosmos SDK MultiStore.
+2. Ethermint CommitStateDB.
+
+Cosmos SDK MultiStore is used in most Ethermint modules such as bank, governance and staking. The synchronization of this store on the network is taken care of by [Tendermint](https://github.com/tendermint/tendermint).
+
+A unique store prefix(also referred to as the store key) will be assigned to each Ethermint module since they all implement the cosmos-sdk MultiStore interface. Cosmos MultiStore can be accessed by `sdk.Context` expect `ethermint/evm` module.
+
+For `evm` module, Ethermint creates a type called `CommitStateDB` which implements `ethvm.StateDB` interface from go-ethereum, and it aims to be compatible with ethereum EVM transactions.
+
+## Database
```
Comment (araskachoi) [inline docs/core/store.md:28]: maybe this would be a good place to maybe add some information about the IAVL tree vs Merkle here.

**Review by araskachoi — COMMENTED** [review]
looks good, just a few recommendations

**Comment by araskachoi** [comment]
@freddyli7 can we make the recommended changes to merge?

**Pull request closed without merging**